### PR TITLE
Fix confirmation modal's confirm button not disabled after clicking

### PR DESCRIPTION
**Pull request by shanggeeth**
### Purpose
This will fix the following issues
- Confirmation modal's primary action button not disabled when primary action is loading
- Add logic to disable confirmation modal's confirm button when clicked until the deletion is done for,
-- deleting an application from `Applications` in `Console`.
--deleting an identity provider from that identity provider's `Danger Zone` in `Console`.
-- deleting an identity provider from `Identity Providers` in `Console`.

Before

https://user-images.githubusercontent.com/59327626/189600336-11d5dbee-c6b1-47ea-b387-239a7910d0a2.mov

After

https://user-images.githubusercontent.com/59327626/189598796-4e8d0d71-bbaf-489a-917c-80047b309323.mov

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/applications/components/application-list.tsx [hunk 1]
```diff
@@ -543,7 +545,12 @@ export const ApplicationList: FunctionComponent<ApplicationListPropsInterface> =
                             setShowDeleteConfirmationModal(false);
                             setAlert(null);
                         } }
-                        onPrimaryActionClick={ (): void => handleApplicationDelete(deletingApplication.id) }
+                        onPrimaryActionClick={ (): void => {
+                            setLoading(true);
+                            handleApplicationDelete(deletingApplication.id).finally(() => {
```
Comment (kavinda1995) [inline apps/console/src/features/applications/components/application-list.tsx:550]: We can extract this callback's inner function calls (setLoading and .finally() part) and place them inside **handleApplicationDelete** function. So we can have containerised handleApplicationDelete function, which can be used in other places too without duplicating the logic of handling setLoading, thus can remove the promise return from the **handleApplicationDelete** function as well

File: apps/console/src/features/identity-providers/components/identity-provider-list.tsx [hunk 2]
```diff
@@ -440,9 +442,12 @@ export const IdentityProviderList: FunctionComponent<IdentityProviderListPropsIn
                         primaryAction={ t("common:confirm") }
                         secondaryAction={ t("common:cancel") }
                         onSecondaryActionClick={ (): void => setShowDeleteConfirmationModal(false) }
-                        onPrimaryActionClick={
-                            (): void => handleIdentityProviderDelete(deletingIDP.id)
-                        }
+                        onPrimaryActionClick={ (): void => {
+                            setLoading(true);
+                            handleIdentityProviderDelete(deletingIDP.id).finally(() => {
```
Comment (kavinda1995) [inline apps/console/src/features/identity-providers/components/identity-provider-list.tsx:447]: We can abstract this function too as in the above comment

File: apps/console/src/features/identity-providers/components/settings/general-settings.tsx [hunk 3]
```diff
@@ -362,9 +364,12 @@ export const GeneralSettings: FunctionComponent<GeneralSettingsInterface> = (
                                 primaryAction={ t("common:confirm") }
                                 secondaryAction={ t("common:cancel") }
                                 onSecondaryActionClick={ (): void => setShowDeleteConfirmationModal(false) }
-                                onPrimaryActionClick={
-                                    (): void => handleIdentityProviderDelete()
-                                }
+                                onPrimaryActionClick={ (): void => {
+                                    setLoading(true);
+                                    handleIdentityProviderDelete().finally(() => {
```
Comment (kavinda1995) [inline apps/console/src/features/identity-providers/components/settings/general-settings.tsx:369]: Abstracting the function can be done as same as the above comment

File: apps/console/src/features/applications/components/application-list.tsx (hunk 1 above)
Comment (shanggeeth) [reply apps/console/src/features/applications/components/application-list.tsx]: Addressed via af36e121eed12e6c0f06003fee7e01072b7c4ee6

File: apps/console/src/features/identity-providers/components/identity-provider-list.tsx (hunk 2 above)
Comment (shanggeeth) [reply apps/console/src/features/identity-providers/components/identity-provider-list.tsx]: Addressed via af36e121eed12e6c0f06003fee7e01072b7c4ee6

File: apps/console/src/features/identity-providers/components/settings/general-settings.tsx (hunk 3 above)
Comment (shanggeeth) [reply apps/console/src/features/identity-providers/components/settings/general-settings.tsx]: Addressed via af36e121eed12e6c0f06003fee7e01072b7c4ee6

**Review by DonOmalVindula — APPROVED** [review]

**Pull request merged**